ci/rpm/Dockerfile

```diff
@@ -1,17 +1,25 @@
-FROM fedora:31
+FROM fedora:31 as base
 SHELL ["/bin/bash", "-e", "-o", "pipefail", "-u", "-x", "-c"]
 
 RUN dnf install -y python3-pip rpm-build
 RUN ln -s /usr/bin/python3 /usr/local/bin/python
 
+FROM base AS CIRCLE_CI
+COPY entrypoint.sh /root/
+RUN chmod +x /root/entrypoint.sh
+CMD ["/root/entrypoint.sh"]
+WORKDIR /root/git-machete
+
+FROM base AS LOCAL
 ARG user_id
 ARG group_id
 RUN set -x \
     && [ ${user_id:-0} -ne 0 ] \
     && [ ${group_id:-0} -ne 0 ] \
-    && groupadd -g ${group_id} ci-user \
-    && useradd -l -u ${user_id} -g ci-user ci-user \
-    && install -d -m 0755 -o ci-user -g ci-user /home/ci-user
+    # sometimes the given `group_id` is already taken and `addgroup` raises an error, so let's check its existence first
+    && (getent group $group_id || addgroup --gid=${group_id} ci-user) \
```

Comment:
Whoops

```
Step 13/18 : RUN set -x     && [ ${user_id:-0} -ne 0 ]     && [ ${group_id:-0} -ne 0 ]     && (getent group $group_id || addgroup --gid=${group_id} ci-user)     && useradd -l -u ${user_id} -g $group_id ci-user     && install -d -m 0755 -o ci-user -g $group_id /home/ci-user
 ---> Running in 740b6eef02a2
+ set -x
+ '[' 1000 -ne 0 ']'
+ '[' 1000 -ne 0 ']'
+ getent group 1000
+ addgroup --gid=1000 ci-user
/bin/bash: addgroup: command not found
```

Reply:
The entire change here is probably not needed, can stay as it was

Reply:
Well, i got the error that the group exists and i cannot create it and that's why the change

Reply:
`------                                                                                                                                                                                                                                                                                                                  
 > [local 1/4] RUN set -x     && [ 501 -ne 0 ]     && [ 20 -ne 0 ]     && groupadd -g 20 ci-user     && useradd -l -u 501 -g ci-user ci-user     && install -d -m 0755 -o ci-user -g ci-user /home/ci-user:                                                                                                             
#8 0.299 + set -x                                                                                                                                                                                                                                                                                                       
#8 0.300 + '[' 501 -ne 0 ']'
#8 0.300 + '[' 20 -ne 0 ']'
#8 0.300 + groupadd -g 20 ci-user
#8 0.301 groupadd: GID '20' already exists
`

Reply:
Well, i just executed it for the mentioned commit and it works:
` => CACHED [local 1/4] RUN set -x     && [ 501 -ne 0 ]     && [ 20 -ne 0 ]     && (getent group 20 || addgroup --gid=20 ci-user)     && useradd -l -u 501 -g 20 ci-user     && install -d -m 0755 -o ci-user -g 20 /home/ci-user `

Reply:
Sth is wrong, your message should start with: `[local 1/4] RUN set`
but it starts with: `Step 13/18 : RUN set`. Can you run it again, this time from the newest commit: [7d83000](https://github.com/VirtusLab/git-machete/pull/422/commits/7d8300008edcba23b751df37fc76aca0c62513f4)

Reply:
Oh, I see where the problem lies... our environments still do differ slightly:

In your case, host GID is 20, which happens to correspond to a group already existing in `fedora:35`, so `addgroup` is NOT executed. 

In my case, host GID is 1000, which does NOT correspond to  group already existing in `fedora:35`, so `addgroup` is attempted but fails, since such executable does NOT exist on fedora:

```
$ docker run fedora:35 groupadd --help
Usage: groupadd [options] GROUP

Options:
.....
```

but:

```
docker run fedora:35 addgroup --help
docker: Error response from daemon: OCI runtime create failed: container_linux.go:380: starting container process caused: exec: "addgroup": executable file not found in $PATH: unknown.
ERRO[0000] error waiting for container: context canceled 
```